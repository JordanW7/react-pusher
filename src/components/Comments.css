.comments {
	height: 403px;
	min-width: 200px;
	background: rgba(0, 0, 0, 0.2);
	color: white;
	border: 1px solid rgba(0, 0, 0, 0.2);
}

.comments-current {
	overflow: scroll;
	max-height: 355px;
}

.comments-button {
	margin-left: 5px;
}

.comments-input {
	width: 100%;
    max-width: 1000px;
    box-sizing: border-box;
	margin-left: 5px;
}

.comments-form {
	align-items: center;
	justify-content: center;
}
