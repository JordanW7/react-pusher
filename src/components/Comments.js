import React, { Component } from "react";
import { Row, Col, Input, Button } from "antd";
import CommentEntry from "./CommentEntry";
import { subscribeToTimer } from "../api/api";
import "./Comments.css";

class Comments extends Component {
  constructor(props) {
    super(props);
    this.state = {
      timestamp: "no timestamp yet"
    };
    subscribeToTimer((err, timestamp) =>
      this.setState({
        timestamp
      })
    );
  }
  render() {
    console.log("STAMP", this.state.timestamp);
    return (
      <section className="comments">
        <div className="comments-current">
          <CommentEntry />
          <CommentEntry />
          <CommentEntry />
          <CommentEntry />
          <CommentEntry />
        </div>
        <Row>
          <Col xs={21} sm={21} md={21} lg={18} xl={18}>
            <Input className="comments-input" />
          </Col>
          <Col xs={3} sm={3} md={3} lg={6} xl={6}>
            <Button
              type="primary"
              onClick={this.handleSubmit}
              className="comments-button"
            >
              Add
            </Button>
          </Col>
        </Row>
      </section>
    );
  }
}

export default Comments;
