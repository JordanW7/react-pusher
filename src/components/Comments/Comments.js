import React, { Component } from "react";
import { Row, Col, Input, Button, message } from "antd";
import CommentEntry from "./CommentEntry/CommentEntry";
import { subscribeToComments, submitComment } from "../../api/api";
import "./Comments.css";

class Comments extends Component {
  constructor(props) {
    super(props);
    this.state = {
      comments: [],
      usercomment: ""
    };
    try {
      subscribeToComments((err, newcomment) => {
        this.setState(prevState => ({
          comments: [newcomment, ...prevState.comments]
        }));
      });
    } catch (err) {
      message.error("Error subscribing to comments");
    }
  }
  onMessageChange = event => {
    this.setState({ usercomment: event.target.value });
  };
  onKeyPress = e => {
    if (e.which === 13) {
      this.onMessageSubmit();
    }
  };
  onMessageSubmit = () => {
    const { usercomment } = this.state;
    if (!usercomment) {
      return message.error("Please provide a message to submit");
    }
    try {
      submitComment(usercomment);
      const newcomment = ["You", "", "", usercomment, new Date()];
      this.setState(prevState => ({
        comments: [newcomment, ...prevState.comments]
      }));
    } catch (err) {
      message.error("Error submitting comment");
    }
  };
  render() {
    return (
      <section className="comments">
        <div className="comments-current">
          {this.state.comments.map((result, i) => {
            return (
              <CommentEntry
                key={`${result[0]} ${result[1]} ${result[4]}`}
                name={`${result[0]} ${result[1]}`}
                img={result[2]}
                comment={result[3]}
                time={result[4]}
              />
            );
          })}
        </div>
        <div className="comments-addbox">
          <Row>
            <Col xs={21} sm={21} md={21} lg={18} xl={18}>
              <label htmlFor="addcommentfield" style={{ display: "none" }}>
                Add a comment
              </label>
              <Input
                className="comments-input"
                onChange={this.onMessageChange}
                onKeyPress={this.onKeyPress}
                placeholder="Add a comment"
                id="addcommentfield"
              />
            </Col>
            <Col xs={3} sm={3} md={3} lg={6} xl={6}>
              <Button
                id="addcommentsubmit"
                type="primary"
                onClick={this.onMessageSubmit}
                className="comments-button"
              >
                Add
              </Button>
            </Col>
          </Row>
        </div>
      </section>
    );
  }
}

export default Comments;
