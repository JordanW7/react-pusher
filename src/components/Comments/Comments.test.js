import React from "react";
import { shallow } from "enzyme";
import Comments from "./Comments";

it("expect to render Comments component", () => {
  const wrapper = shallow(<Comments />);
  expect(wrapper).toMatchSnapshot();
});

it("Correctly submits new comment", () => {
  const wrapper = shallow(<Comments />);
  wrapper.find('[id="addcommentsubmit"]').simulate("click");
  expect(wrapper.state()).toEqual({
    comments: [],
    usercomment: ""
  });
});

it("Correctly updates comment state", () => {
  const wrapper = shallow(<Comments />);
  wrapper
    .find('[id="addcommentfield"]')
    .simulate("change", { target: { value: "Hello" } });
  expect(wrapper.state()).toEqual({
    comments: [],
    usercomment: "Hello"
  });
});
