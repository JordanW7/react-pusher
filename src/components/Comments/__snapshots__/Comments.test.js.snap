// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Comments component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Comments />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="comments-current"
        />,
        <div
          className="comments-addbox"
        >
          <Row
            gutter={0}
          >
            <Col
              lg={18}
              md={21}
              sm={21}
              xl={18}
              xs={21}
            >
              <label
                htmlFor="addcommentfield"
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                Add a comment
              </label>
              <Input
                className="comments-input"
                disabled={false}
                id="addcommentfield"
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder="Add a comment"
                prefixCls="ant-input"
                type="text"
              />
            </Col>
            <Col
              lg={6}
              md={3}
              sm={3}
              xl={6}
              xs={3}
            >
              <Button
                block={false}
                className="comments-button"
                ghost={false}
                id="addcommentsubmit"
                loading={false}
                onClick={[Function]}
                prefixCls="ant-btn"
                type="primary"
              >
                Add
              </Button>
            </Col>
          </Row>
        </div>,
      ],
      "className": "comments",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "comments-current",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Row
            gutter={0}
          >
            <Col
              lg={18}
              md={21}
              sm={21}
              xl={18}
              xs={21}
            >
              <label
                htmlFor="addcommentfield"
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                Add a comment
              </label>
              <Input
                className="comments-input"
                disabled={false}
                id="addcommentfield"
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder="Add a comment"
                prefixCls="ant-input"
                type="text"
              />
            </Col>
            <Col
              lg={6}
              md={3}
              sm={3}
              xl={6}
              xs={3}
            >
              <Button
                block={false}
                className="comments-button"
                ghost={false}
                id="addcommentsubmit"
                loading={false}
                onClick={[Function]}
                prefixCls="ant-btn"
                type="primary"
              >
                Add
              </Button>
            </Col>
          </Row>,
          "className": "comments-addbox",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Col
                lg={18}
                md={21}
                sm={21}
                xl={18}
                xs={21}
              >
                <label
                  htmlFor="addcommentfield"
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                >
                  Add a comment
                </label>
                <Input
                  className="comments-input"
                  disabled={false}
                  id="addcommentfield"
                  onChange={[Function]}
                  onKeyPress={[Function]}
                  placeholder="Add a comment"
                  prefixCls="ant-input"
                  type="text"
                />
              </Col>,
              <Col
                lg={6}
                md={3}
                sm={3}
                xl={6}
                xs={3}
              >
                <Button
                  block={false}
                  className="comments-button"
                  ghost={false}
                  id="addcommentsubmit"
                  loading={false}
                  onClick={[Function]}
                  prefixCls="ant-btn"
                  type="primary"
                >
                  Add
                </Button>
              </Col>,
            ],
            "gutter": 0,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <label
                    htmlFor="addcommentfield"
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  >
                    Add a comment
                  </label>,
                  <Input
                    className="comments-input"
                    disabled={false}
                    id="addcommentfield"
                    onChange={[Function]}
                    onKeyPress={[Function]}
                    placeholder="Add a comment"
                    prefixCls="ant-input"
                    type="text"
                  />,
                ],
                "lg": 18,
                "md": 21,
                "sm": 21,
                "xl": 18,
                "xs": 21,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Add a comment",
                    "htmlFor": "addcommentfield",
                    "style": Object {
                      "display": "none",
                    },
                  },
                  "ref": null,
                  "rendered": "Add a comment",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "comments-input",
                    "disabled": false,
                    "id": "addcommentfield",
                    "onChange": [Function],
                    "onKeyPress": [Function],
                    "placeholder": "Add a comment",
                    "prefixCls": "ant-input",
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Button
                  block={false}
                  className="comments-button"
                  ghost={false}
                  id="addcommentsubmit"
                  loading={false}
                  onClick={[Function]}
                  prefixCls="ant-btn"
                  type="primary"
                >
                  Add
                </Button>,
                "lg": 6,
                "md": 3,
                "sm": 3,
                "xl": 6,
                "xs": 3,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "block": false,
                  "children": "Add",
                  "className": "comments-button",
                  "ghost": false,
                  "id": "addcommentsubmit",
                  "loading": false,
                  "onClick": [Function],
                  "prefixCls": "ant-btn",
                  "type": "primary",
                },
                "ref": null,
                "rendered": "Add",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="comments-current"
          />,
          <div
            className="comments-addbox"
          >
            <Row
              gutter={0}
            >
              <Col
                lg={18}
                md={21}
                sm={21}
                xl={18}
                xs={21}
              >
                <label
                  htmlFor="addcommentfield"
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                >
                  Add a comment
                </label>
                <Input
                  className="comments-input"
                  disabled={false}
                  id="addcommentfield"
                  onChange={[Function]}
                  onKeyPress={[Function]}
                  placeholder="Add a comment"
                  prefixCls="ant-input"
                  type="text"
                />
              </Col>
              <Col
                lg={6}
                md={3}
                sm={3}
                xl={6}
                xs={3}
              >
                <Button
                  block={false}
                  className="comments-button"
                  ghost={false}
                  id="addcommentsubmit"
                  loading={false}
                  onClick={[Function]}
                  prefixCls="ant-btn"
                  type="primary"
                >
                  Add
                </Button>
              </Col>
            </Row>
          </div>,
        ],
        "className": "comments",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "comments-current",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Row
              gutter={0}
            >
              <Col
                lg={18}
                md={21}
                sm={21}
                xl={18}
                xs={21}
              >
                <label
                  htmlFor="addcommentfield"
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                >
                  Add a comment
                </label>
                <Input
                  className="comments-input"
                  disabled={false}
                  id="addcommentfield"
                  onChange={[Function]}
                  onKeyPress={[Function]}
                  placeholder="Add a comment"
                  prefixCls="ant-input"
                  type="text"
                />
              </Col>
              <Col
                lg={6}
                md={3}
                sm={3}
                xl={6}
                xs={3}
              >
                <Button
                  block={false}
                  className="comments-button"
                  ghost={false}
                  id="addcommentsubmit"
                  loading={false}
                  onClick={[Function]}
                  prefixCls="ant-btn"
                  type="primary"
                >
                  Add
                </Button>
              </Col>
            </Row>,
            "className": "comments-addbox",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Col
                  lg={18}
                  md={21}
                  sm={21}
                  xl={18}
                  xs={21}
                >
                  <label
                    htmlFor="addcommentfield"
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  >
                    Add a comment
                  </label>
                  <Input
                    className="comments-input"
                    disabled={false}
                    id="addcommentfield"
                    onChange={[Function]}
                    onKeyPress={[Function]}
                    placeholder="Add a comment"
                    prefixCls="ant-input"
                    type="text"
                  />
                </Col>,
                <Col
                  lg={6}
                  md={3}
                  sm={3}
                  xl={6}
                  xs={3}
                >
                  <Button
                    block={false}
                    className="comments-button"
                    ghost={false}
                    id="addcommentsubmit"
                    loading={false}
                    onClick={[Function]}
                    prefixCls="ant-btn"
                    type="primary"
                  >
                    Add
                  </Button>
                </Col>,
              ],
              "gutter": 0,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <label
                      htmlFor="addcommentfield"
                      style={
                        Object {
                          "display": "none",
                        }
                      }
                    >
                      Add a comment
                    </label>,
                    <Input
                      className="comments-input"
                      disabled={false}
                      id="addcommentfield"
                      onChange={[Function]}
                      onKeyPress={[Function]}
                      placeholder="Add a comment"
                      prefixCls="ant-input"
                      type="text"
                    />,
                  ],
                  "lg": 18,
                  "md": 21,
                  "sm": 21,
                  "xl": 18,
                  "xs": 21,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Add a comment",
                      "htmlFor": "addcommentfield",
                      "style": Object {
                        "display": "none",
                      },
                    },
                    "ref": null,
                    "rendered": "Add a comment",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "className": "comments-input",
                      "disabled": false,
                      "id": "addcommentfield",
                      "onChange": [Function],
                      "onKeyPress": [Function],
                      "placeholder": "Add a comment",
                      "prefixCls": "ant-input",
                      "type": "text",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Button
                    block={false}
                    className="comments-button"
                    ghost={false}
                    id="addcommentsubmit"
                    loading={false}
                    onClick={[Function]}
                    prefixCls="ant-btn"
                    type="primary"
                  >
                    Add
                  </Button>,
                  "lg": 6,
                  "md": 3,
                  "sm": 3,
                  "xl": 6,
                  "xs": 3,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "block": false,
                    "children": "Add",
                    "className": "comments-button",
                    "ghost": false,
                    "id": "addcommentsubmit",
                    "loading": false,
                    "onClick": [Function],
                    "prefixCls": "ant-btn",
                    "type": "primary",
                  },
                  "ref": null,
                  "rendered": "Add",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
